node/lib/cmd/commit.js

```diff
@@ -270,6 +332,10 @@ const doAmend = co.wrap(function *(args) {
  * @param {String}  [args.message]
  */
 exports.executeableSubcommand = function (args) {
+    if (args.all && 1 !== args.file.length) {
```

Comment:
Here, you check 1 !== args.file.length, but further up, you check 0 == args.file.length.  Which is it?